python/src/tstools/unrFiles.py

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env python3
+
+"""
+Module for downloading and handling UNR generated files.
+"""
+
+import requests as req
+import numpy as np
+
+from tstools import inputFileIO as ifio
+from tstools.util.convtime import convtime, yy_to_yyyy
+from tstools.util.nutils import msg_err
+
+########################################################################
+"""
+Constants
+"""
+
+ALL = 'all'
+BLANK_STR = '' 
+
+EQUIP_CHNG_CODE = 1
+EQ_CODE = 2
+
+JAN, FEB, MAR, APR, MAY = ['JAN','FEB','MAR','APR','MAY']
+JUN, JUL, AUG, SEP, OCT = ['JUN','JUL','AUG','SEP','OCT']
+NOV, DEC = ['NOV','DEC']
+
+########################################################################
+class UnrTxyz2:
+
+    """
+    Download, read in local copy, modify, or write out UNR-style .txyz2
+    position time series files.
+    """
+
+    ####################################################################
+    def __init__(self, site=''):
+
+        self.baseUrl = ('http://geodesy.unr.edu/gps_timeseries/txyz/'
+                       +'IGS08/')
+        if site != BLANK_STR:
+            site = site.upper()
+            self.site = site
+        else:
+            self.site = BLANK_STR
+        self.url = BLANK_STR
+        self.lineList = []
+
+    ####################################################################
+    def fetch(self, site='', baseUrl=''):
+
+        """
+        Read UNR .txyz2 position time series file from UNR website for
+        specified station.
+        """
+
+        if baseUrl != BLANK_STR:
+            self.baseUrl = baseUrl
+
+        if site != BLANK_STR:
+            site = site.upper()
+            self.site = site
+        
+        self.url = f'{self.baseUrl}{self.site}.IGS08.txyz2'
+        
+        # download file
+        raw = req.get(self.url)
+
+        # split lines into list elements
+        for line in raw.text.split('\n'):
+            # if self.site not 'all' only pull out the lines
+            # for the site of interest
+            if self.site == ALL:
+                self.lineList.append(line)
+            else:
+                if line == BLANK_STR:
+                    continue
+                if line.split()[0] == self.site:
+                    self.lineList.append(line)
+
+    ####################################################################
+    def write(self, fileName=''):
+
+        """
+        Write UnrTxyz2 object to .txyz2 formatted text file.
+        """
+
+        if fileName == BLANK_STR:
+            fileName = f'./{self.site}.IGS08.txyz2'
+            
+        wf = open(fileName,'w')
+        for line in self.lineList:
+            if line == BLANK_STR:
+                continue
+            if line.split()[0] == self.site:
+                wf.write(line+'\n')
+
+########################################################################
+class UnrLatLonHtFile:
```

Comment:
Looks like the following classes are all very similar to the first class in this file. It might be a good example of using some inheritance to take care of the generic work that's being done.